apis/v1alpha1/httproute_types.go

```diff
@@ -267,15 +265,17 @@ type HTTPHeaderMatch struct {
 	// Please read the implementation's documentation to determine the supported
 	// dialect.
 	//
+	// HTTP Header name matching MUST be case-insensitive (RFC 2616 - section 4.2).
+	//
 	// +kubebuilder:default=Exact
 	Type HeaderMatchType `json:"type,omitempty"`
 
 	// Values is a map of HTTP Headers to be matched.
 	// It MUST contain at least one entry.
 	//
 	// The HTTP header field name to match is the map key, and the
-	// value of the HTTP header is the map value. HTTP header field
-	// names MUST be matched case-insensitively.
+	// value of the HTTP header is the map value. HTTP header field names matching
```

Comment:
This is _super_ nitpicky, but I think it should be `name` instead of `names` here.